src/test/java/teammates/test/pageobjects/InstructorFeedbackResultsPage.java

```diff
@@ -372,10 +376,36 @@ public void hoverClickAndViewPhotoOnTableCell(int questionBodyIndex, int tableRo
                               + " tr:nth-child(" + (tableRow + 1) + ")"
                               + " td:nth-child(" + (tableCol + 1) + ")";
         String popoverSelector = cellSelector + " .popover-content";
+        String hoverSelector = cellSelector + " .profile-pic-icon-hover";
 
-        moveToElement(By.cssSelector(cellSelector + " .profile-pic-icon-hover"));
+        moveToElement(By.cssSelector(hoverSelector));
         waitForElementPresence(By.cssSelector(popoverSelector + " > a")).click();
 
+        verifyPopoverImageUrlWithHoverRetry(popoverSelector, hoverSelector, urlRegex,
+                "Hover and verify photo on table cell");
+    }
+
+    /**
+     * Popovers triggered by hover actions sometimes fail to appear, resulting in a {@link WebDriverException}.
+     * Popover image verifications that depend on hover actions should therefore be retried several times,
+     * with the hover action triggered before each retry.
+     */
+    private void verifyPopoverImageUrlWithHoverRetry(
+            final String popoverSelector, final String hoverSelector, final String urlRegex, String taskName) {
+        RetryManager.runUntilNoException(new RetryableTask(taskName) {
+            @Override
+            public void run() {
+                verifyPopoverImageUrl(popoverSelector, urlRegex);
+            }
+
+            @Override
+            public void beforeRetry() {
+                moveToElement(By.cssSelector(hoverSelector));
```

Comment:
Well, it seems to work (I've run it many times and on several runs, the retry does get triggered, and the first retry works).